scripts/support/runner/openocd.py

```diff
@@ -94,8 +94,12 @@ def create_from_env(command, debug):
         zephyr_base = get_env_or_bail('ZEPHYR_BASE')
         arch = get_env_or_bail('ARCH')
         board_name = get_env_or_bail('BOARD_NAME')
-        openocd_config = path.join(zephyr_base, 'boards', arch,
-                                   board_name, 'support', 'openocd.cfg')
+        for dirpath, dirnames, filenames in os.walk(path.join(zephyr_base, 'boards', arch),
```

Comment:
I'm not sure what this is trying to do (the commit message is pretty skimpy); would you mind explaining? This seems to have some problems.

In particular:

1. do you really want to use os.walk()? That will look for $BOARD.yaml in every directory under boards/$ARCH/, including the doc/ directories, etc.
2. Won't this fail in a confusing way (to the user) if there is no such yaml file?
3. What is the correct behavior if there are multiple candidate yaml files?
4. This is duplicated code, why not add it as a method to ZephyrBinaryRunner in core.py?

It seems like you might want something like this instead (in a new method in ZephyrBinaryRunner):

```
board_yaml_files = glob.glob(path.join(zephyr_base, 'boards', arch, '*',
                                       '{}.yaml'.format(board_name)))
if not board_yaml_files:
    raise ValueError('no {}.yaml was found'.format(board_name))

board_dir = path.dirname(board_yaml_files[0])
openocd_config = path.join(board_dir, 'support', 'openocd.cfg')
```

But that just takes the directory from the first found YAML file, which might not be what you want (though it's what this code is doing).

Reply:
yeah, above code works better I guess.
I still need to wrap my header around those scripts and understand them better :)

Reply:
On second thought, since it's openocd-specific, maybe a freestanding function in core.py instead of a method in the runner.

Reply:
I found another easier way to do that and cleaning up and removing the need for some variables. cmake already finds the BOARD_DIR for us, so no reason to search for it again in the flasher